code/modules/mob/living/carbon/human/human_attackhand.dm

```diff
@@ -155,8 +155,8 @@
 						return held_weapon.afterattack(target,src)
 
 			var/disarm_chance = rand(1, 100)
-			var/attacker_skill_level = skills && attacking_mob.skills ? skills.get_skill_level(SKILL_CQC) : SKILL_CQC_MAX // No skills, so assume max
-			var/defender_skill_level = skills ? skills.get_skill_level(SKILL_CQC) : SKILL_CQC_MAX // No skills, so assume max
+			var/attacker_skill_level = attacking_mob.skills && skills ? attacking_mob.skills.get_skill_level(SKILL_CQC) : SKILL_CQC_MAX // No skills, so assume max
+			var/defender_skill_level = skills && attacking_mob.skills ? skills.get_skill_level(SKILL_CQC) : SKILL_CQC_MAX // No skills, so assume max
```

Comment:
Either I don't understand what you're trying to do by doing it this way, but I would think you are merely wanting to either get each mob's skill level or if they don't have skills to assume max. I don't know why whether the src having skill or not has to be taken into account when determining the skill level of two different mobs.

```suggestion
			var/attacker_skill_level = attacking_mob.skills ? attacking_mob.skills.get_skill_level(SKILL_CQC) : SKILL_CQC_MAX // No skills, so assume max
			var/defender_skill_level = skills ? skills.get_skill_level(SKILL_CQC) : SKILL_CQC_MAX // No skills, so assume max
```

Reply:
@Drulikar I think it would be better if either the attacker or defender has an undefined CQC skill, to assume max for both participants, since something is a bit strange for that situation to happen. If the attacker or defender is meant to have a CQC advantage, it should be defined. Even in on-the-fly events where someone is meant to have a CQC advantage, it could be defined by admin variable edit in-game, I assume. That's why I wrote it that way.

Reply:
So think again what Fira mentioned:
> assuming maximum if no skills is the intended behavior AFAIK, it's only used for things like admin-set infinite skills. having no skills allow you to bypass skillchecks.

So you're saying an admin set infinite skill person disarming say a rifleman who does have skills set, we need to assume now the rifleman has max SKILL_CQC because one of the parties has no skills set?

Reply:
I took what Fira said to mean "Sometimes admins spawn characters with infinite skills to avoid headaches with them not being able to, say, open a pill bottle, or build a cade, or fly a ship, etc. which is different then "I purposefully want this character to be godly at CQC to the point of being immune to being disarm-stunned - I thought they would be specifically setting the CQC level to max in that case.

Reply:
I still don't see how one party missing skills should result in the party not missing skills returning a different value for their skills.

Reply:
I look at it as "If one party is missing skills, assume an equal CQC level only for purposes of disarms between them" because in the case I mentioned earlier, did the admin intend for the person they spawned to be immune to disarm stuns? And intend for 50% of the disarms they do to stun for 14 seconds each time? If they intended that much of a massive difference to the standard disarm values, why did they not specifically define the CQC level?

Reply:
If that was the intention you'd just put the other person's skill level as the missing person's skills so they have matching skills.

Really think about what you are doing and evaluate how poor this code is as written.

Reply:
Agree with @Drulikar on this one. It makes no sense to me to set the defender's skill based on the attacker's skill. Treat each mob independently.

Reply:
Yes you're both right, my bad for stamping this approved here